diffrax/_integrate.py

```diff
@@ -84,12 +87,20 @@ class State(eqx.Module):
     dense_infos: Optional[BufferDenseInfos]
     dense_save_index: Optional[IntScalarLike]
     progress_meter_state: PyTree[Array]
+    event_result: Optional[PyTree[Union[BoolScalarLike, RealScalarLike]]] = None
+    event_mask: Optional[PyTree[BoolScalarLike]] = None
+    dense_info_for_event: Optional[DenseInfo] = None
+    tprevprev: Optional[FloatScalarLike] = None
```

Comment:
I don't think default arguments are needed here.

(This is also just a general good-practice-rule-of-thumb: try to avoid default arguments where possible, as they're a common source of unexpected behaviour. So in practice this usually means putting them on public interfaces.)